ASL-716 accessibility enhancements phase 11: examples and fixes (#1233)
herd/herdtools7: asllib/doc/Specifications.tex

\]
returns the set of dependent pairs of identifiers $\dependencies$ induced by the declaration $\vd$.

> \ExampleDef{The Dependencies Global Declarations}

Reviewer: ```suggestion
\ExampleDef{The Dependencies of Global Declarations}
```

\]
returns the identifier $\name$ being defined by the declaration $\vd$.

\ExampleDef{The Identifiers Defined by Global Declarations}
The specification given in \listingref{DefDecl},

> shows examples of global declarations and the identifiers the use,

Reviewer: ```suggestion
shows examples of global declarations and the identifiers they define,
```

\]
returns the set of identifiers $\ids$ which the declaration $\vd$ depends on.

\ExampleDef{Identifiers Used by Global Declarations}
The specification in \listingref{UseDecl} shows the set of identifiers
used by each global declaration via comments above it.

> \ASLListing{Identifiers used by global decalrations}{UseDecl}{\typingtests/TypingRule.UseDecl.asl}

Reviewer: ```suggestion
\ASLListing{Identifiers used by global declarations}{UseDecl}{\typingtests/TypingRule.UseDecl.asl}
```

\hypertarget{def-usesubtypes}{}
The function
\[
\usesubtypes(\overname{\langle(\overname{\identifier}{\vx}\times\overname{\Field^*}{\subfields})\rangle}{\fields}) \aslto \overname{\pow{\defusename}}{\ids}
\]
returns the set of identifiers $\ids$ which the \optional\ pair consisting of
identifier $\vx$ (the type being subtyped) and fields $\subfields$ depends on.

> \ExampleDef{Identifiers Used by a Subtyping Declaration}

Reviewer: (consistency)
```suggestion
\ExampleDef{The Identifiers Used by a Subtyping Declaration}
```

\]
returns the set of identifiers $\ids$ which the expression or \optional\ expression $\ve$ depends on.

> \ExampleDef{Identifiers Used by Expressions}

Reviewer: (consistency)
```suggestion
\ExampleDef{The Identifiers Used by Expressions}
```
